pkg/controller/deployment/util/deployment_util.go

```diff
@@ -835,3 +835,22 @@ func (o BySelectorLastUpdateTime) Less(i, j int) bool {
 	}
 	return ti.Before(tj)
 }
+
+// OverlapsWith returns true when two given deployments are different and overlap with each other
+func OverlapsWith(current, other *extensions.Deployment) bool {
+	if current.UID == other.UID {
+		return false
+	}
+	currentSelector, err := unversioned.LabelSelectorAsSelector(current.Spec.Selector)
+	if err != nil {
+		glog.Errorf("Skip overlapping check: deployment %s/%s has invalid label selector: %v", current.Namespace, current.Name, err)
```

Comment:
It's fine to error out here since this is the current selector. Just not in the other selector.